src/engine/controls/bpmcontrol.cpp

```diff
@@ -442,23 +445,32 @@ double BpmControl::calcSyncAdjustment(double my_percentage, bool userTweakingSyn
     // than modular 1.0 beat fractions. This will allow sync to work across loop
     // boundaries too.
 
+
     double master_percentage = m_dSyncTargetBeatDistance.getValue();
+
+    // If we're doing half bpm adjustment, the master beat distance will never get over 0.5.
+    // We have to halve our beat distance to do the sync adjustment calculation.
+    double adjust_factor = m_dSyncAdjustFactor.getValue();
+    if (adjust_factor == 0.5) {
+        while (my_percentage >= 0.5) {
+            my_percentage -= 0.5;
+        }
+    }
     double shortest_distance = shortestPercentageChange(
```

Comment:
This means, this one is also off by one buffer, because the master runs first and has likely already updated to the new distance, while the current buffer has not yet been processed.

Reply:
not quite -- the master doesn't update its new distance until postProcess is called.  This code is called on all decks before postProcess so no distances have been updated yet.